src/emulator/gui.ts

```diff
@@ -0,0 +1,50 @@
+import { EmulatorInstance, EmulatorInfo, Emulators } from "./types";
+import * as javaEmulators from "../serve/javaEmulators";
+import { EmulatorRegistry } from "./registry";
+import { DatabaseEmulator } from "./databaseEmulator";
+import { FirestoreEmulator } from "./firestoreEmulator";
+import * as url from "url";
+import { EmulatorHub } from "./hub";
+
+export interface EmulatorGUIOptions {
+  port: number;
+  host: string;
+  hubInfo: EmulatorInfo;
+  projectId: string;
+  auto_download?: boolean;
+}
+
+export class EmulatorGUI implements EmulatorInstance {
+  constructor(private args: EmulatorGUIOptions) {}
+
+  start(): Promise<void> {
+    const { hubInfo, auto_download } = this.args;
+    const env: NodeJS.ProcessEnv = {
+      HOST: this.args.host.toString(),
+      PORT: this.args.port.toString(),
+      GCLOUD_PROJECT: this.args.projectId,
+      [EmulatorHub.EMULATOR_HUB_ENV]: `${hubInfo.host}:${hubInfo.port}`,
+    };
+
+    return javaEmulators.start(Emulators.GUI, { auto_download }, env);
```

Comment:
Probably out of scope here but seems officially time to rename `javaEmulators` to something like `BinaryEmulators`

Reply:
I'd be happy to follow up with another PR. It's not that hard but it does create a lot of noisy diffs that I want to keep separate.

Reply:
https://github.com/firebase/firebase-tools/pull/1997/